collective/ogtags/adapters.py

```diff
@@ -0,0 +1,36 @@
+from zope.interface import implementer
+
+from collective.ogtags.interfaces import IOGTagsImageProvider
+
+
+@implementer(IOGTagsImageProvider)
+class OGTagsImageDexterityProvider(object):
+
+    def __init__(self, context):
+        self.context = context
+
+    def getOGTagsImage(self):
+        """ Tries to return standard image fields named 'image' and 'lead image'
+            Returns None in case of failure"""
+        for fieldname in ["image", "leadImage"]:
+            try:
+                getattr(self.context, fieldname)
+                return self.context, fieldname
+            except AttributeError:
+                pass
+        return
+
+
+@implementer(IOGTagsImageProvider)
+class OGTagsImageArchetypeProvider(object):
+
+    def __init__(self, context):
+        self.context = context
+
+    def getOGTagsImage(self):
+        """ Tries to return standard image fields named 'image' and 'lead image'
+            Returns None in case of failure"""
+        image_field = self.context.getField("image")
```

Comment:
It would be good to try `leadImage` here as well, for sites that use https://github.com/collective/collective.contentleadimage